Add Video Llava

zucchini-nlp opened the pull request:
# What does this PR do?

Fixes #29640 . Add a new model, Video Llava to the library. 

This is a draft PR, will add more here later.

The model now accepts both, video and image as input in the same batch. Each visual has its own special token, so we do not need to repeat "<image>" 8 times for 8 frames. Here is a short code snippet:  
```python
import torch
import numpy as np
import requests
from PIL import Image
from decord import VideoReader
from huggingface_hub import hf_hub_download
from transformers import VideoLlavaForConditionalGeneration, VideoLlavaProcessor


def load_video_tensor(video_path, n_frms=4, transform=False):
    vr = VideoReader(uri=video_path, height=224, width=224)
    n_frms = min(n_frms, len(vr))
    indices = np.arange(0, len(vr), len(vr) / n_frms).astype(int)
    frames = vr.get_batch(indices).asnumpy()
    return frames

# -------------------------------------------------------------------------------------------------------------------

model = VideoLlavaForConditionalGeneration.from_pretrained("/home/raushan/video_llava/")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

video_path = hf_hub_download(repo_id="raushan-testing-hf/videos-test", filename="sample_demo_1.mp4", repo_type="dataset")
clip = load_video_tensor(video_path, n_frms=8)

processor = VideoLlavaProcessor.from_pretrained("/home/raushan/video_llava")
prompt_vid = "USER: <video>What do you see? ASSISTANT: A baby with a remote control. USER: <video>Why is this funny? ASSISTANT:"
prompt_img = "USER: <image>How many cats are there in the image? ASSISTANT:"
prompt_mix = "USER: <image>How many cats are there in the image? ASSISTANT: 2. USER: <video>Why is this video funny? ASSISTANT:"

inputs = processor(text=[prompt_mix, prompt_img], visual_inputs=[image, clip, image], padding=True, return_tensors="pt")

out = model.generate(**inputs, max_new_tokens=40)
print(processor.batch_decode(out, skip_special_tokens=True, clean_up_tokenization_spaces=True))

# [
#    'USER:  How many cats are there in the image? ASSISTANT: 2. USER:  Why is this video funny? ASSISTANT: The video is funny because the baby is sitting on 
#         the bed and reading a book, which is an unusual and amusing sight. Babies are typically not known for reading books, and the fact', 
#    'USER:  How many cats are there in the image? ASSISTANT: There are two cats in the image..'
# ]


```

zucchini-nlp committed:
add model draft

zucchini-nlp commented:
@LinB203 hey! As we talked before, here is a draft PR of Video Llava. I checked that the modeling part runs without errors and generates similar to the original repo.

To update model files on the hub, you can use `convert_weight` script and use this branch to test if model is loading correctly. Whenever you are available, can you look through, if I missed anything important? :)

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_29733). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

zucchini-nlp committed:
update docstring

zucchini-nlp committed:
add tests

zucchini-nlp committed:
support image and video as input

zucchini-nlp committed:
update for better handling of mixed input and clean-up a bit

zucchini-nlp commented:
@LinB203 pinging in case the first one got lost in notifications :)

ArthurZucker commented:
FYI @NielsRogge  and @amyeroberts

NielsRogge commented on README.md:
```diff
@@ -523,6 +523,7 @@ Current number of checkpoints: ![](https://img.shields.io/endpoint?url=https://h
 1. **[UnivNet](https://huggingface.co/docs/transformers/model_doc/univnet)** (from Kakao Corporation) released with the paper [UnivNet: A Neural Vocoder with Multi-Resolution Spectrogram Discriminators for High-Fidelity Waveform Generation](https://arxiv.org/abs/2106.07889) by Won Jang, Dan Lim, Jaesam Yoon, Bongwan Kim, and Juntae Kim.
 1. **[UPerNet](https://huggingface.co/docs/transformers/model_doc/upernet)** (from Peking University) released with the paper [Unified Perceptual Parsing for Scene Understanding](https://arxiv.org/abs/1807.10221) by Tete Xiao, Yingcheng Liu, Bolei Zhou, Yuning Jiang, Jian Sun.
 1. **[VAN](https://huggingface.co/docs/transformers/model_doc/van)** (from Tsinghua University and Nankai University) released with the paper [Visual Attention Network](https://arxiv.org/abs/2202.09741) by Meng-Hao Guo, Cheng-Ze Lu, Zheng-Ning Liu, Ming-Ming Cheng, Shi-Min Hu.
+1. **[VideoLlava](https://huggingface.co/docs/transformers/main/model_doc/video_llava)** (from YUAN Lab, Peking University) released with the paper [<Video-LLaVA: Learning United Visual Representation by Alignment Before Projection>](<https://arxiv.org/pdf/2311.10122.pdf>) by Bin Lin, Yang Ye, Bin Zhu, Jiaxi Cui, Munang Ning, Peng Jin, Li Yuan.
```
```suggestion
1. **[Video-LLaVA](https://huggingface.co/docs/transformers/main/model_doc/video_llava)** (from YUAN Lab, Peking University) released with the paper [Video-LLaVA: Learning United Visual Representation by Alignment Before Projection](https://arxiv.org/abs/2311.10122) by Bin Lin, Yang Ye, Bin Zhu, Jiaxi Cui, Munang Ning, Peng Jin, Li Yuan.
```
Please link to abstracts instead of pdf :)

NielsRogge commented on src/transformers/models/auto/configuration_auto.py:
```diff
@@ -468,6 +469,7 @@
         ("unispeech-sat", "UNISPEECH_SAT_PRETRAINED_CONFIG_ARCHIVE_MAP"),
         ("univnet", "UNIVNET_PRETRAINED_CONFIG_ARCHIVE_MAP"),
         ("van", "VAN_PRETRAINED_CONFIG_ARCHIVE_MAP"),
+        ("video_llava", "VIDEO_LLAVA_PRETRAINED_CONFIG_ARCHIVE_MAP"),
```
This can be removed everywhere since this is no longer used

zucchini-nlp committed:
bug when mixed inputs & add tests

zucchini-nlp committed:
Update README.md

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

zucchini-nlp commented:
I believe we can start reviewing this now. I converted weights and added them to my hub account temporarily, so that we can run and test the model. 

In the meanwhile will be waiting for @LinB203 to update the weights in the organization hub, which I think is required before we merge the PR

amyeroberts commented:
@zucchini-nlp Awesome work! First thing to do before a full review is resolving the failing tests. Some of these are unrelated, and rebasing on main should resolve. Looking at the CI, some of the failure as video llava specific - let me know if you need any help addressing them!

zucchini-nlp committed:
Merge remote-tracking branch 'upstream/main' into video_llava

zucchini-nlp committed:
link to abstract of paper in README

zucchini-nlp committed:
fix test

zucchini-nlp committed:
fix-copies

zucchini-nlp committed:
Merge branch 'main' into video_llava

zucchini-nlp committed:
make tests happy

zucchini-nlp commented:
Rebased with main and resolved conflicts. The only failing doctest seems to be not able to load and run 7b model in 120sec, but I think we will leave it anyway to show how Video-Llava works

amyeroberts commented:
@zucchini-nlp You can exclude the model from running in the doc tests by adding it to slow_documentation_tests.txt. 

Then, once the PR is reviewed in a steady state and ready for merging, we can run the slow tests and the documentation tests to make sure everything is correct before merging

zucchini-nlp committed:
skip docstest for now

zucchini-nlp committed:
do not run doctest for now

zucchini-nlp committed:
Merge remote-tracking branch 'upstream/main' into video_llava

amyeroberts commented on src/transformers/models/video_llava/image_processing_video_llava.py:
```diff
@@ -0,0 +1,448 @@
+
+    def preprocess(
+        self,
+        images,
+        visual_inputs: List[ImageInput],
```
We might want to define a new type here - as it should be possible to pass in a video as a tensor of size `(batch_size, num_frames, num_channels, height, width)`

```suggestion
        images: ImageInput,
        videos: VideoInput,
```

amyeroberts commented on src/transformers/models/video_llava/modeling_video_llava.py:
```diff
@@ -0,0 +1,667 @@
+    config_class = VideoLlavaConfig
+    base_model_prefix = "model"
+    supports_gradient_checkpointing = True
+    _no_split_modules = ["CLIPAttention"]
```
This should be handled on the size of the vision towers, rather than this model i.e. we shouldn't know which model is loaded, as we're using `AutoModel`

amyeroberts commented on src/transformers/models/video_llava/modeling_video_llava.py:
```diff
@@ -0,0 +1,667 @@
+                module.weight.data[module.padding_idx].zero_()
+
+    @property
+    def _supports_sdpa(self):
+        """
+        Retrieve language_model's attribute to check whether the model supports
+        SDPA or not.
+        """
+        return self.language_model._supports_sdpa
```
~This isn't going to be compatible if we have vision tower's with attention which don't support sdpa. ~

I see below it's handled by passing the attn implementation just to the text tower. I think this is bit confusing  as this value only applies to the text tower and not the whole model.

amyeroberts commented on src/transformers/models/video_llava/modeling_video_llava.py:
```diff
@@ -0,0 +1,667 @@
+        return self.language_model._supports_sdpa
+
+
+VIDEO_LLAVA_INPUTS_DOCSTRING = r"""
```
Missing inputs in docstring

amyeroberts commented on src/transformers/models/video_llava/modeling_video_llava.py:
```diff
@@ -0,0 +1,667 @@
+        >>> from PIL import Image
+        >>> import requests
+        >>> import numpy as np
+        >>> from decord import VideoReader
```
We should use av instead of decord. decord causes crashes on certain GPU setups and isn't compatible with windows

amyeroberts commented on src/transformers/models/video_llava/modeling_video_llava.py:
```diff
@@ -0,0 +1,667 @@
+    def forward(
+        self,
+        input_ids: torch.LongTensor = None,
+        pixel_values_images: torch.FloatTensor = None,
+        pixel_values_videos: torch.FloatTensor = None,
```
It's not ideal having to change the standard input type names. I _think_ this is OK, as we have e.g. `input_ids_query`, `input_ids_doc` for retribert

amyeroberts commented on src/transformers/models/video_llava/modeling_video_llava.py:
```diff
@@ -0,0 +1,667 @@
+        special_image_token_mask = input_ids == special_vision_token
+        num_special_image_tokens = torch.sum(special_image_token_mask, dim=-1)
+        # Compute the maximum embed dimension
+        max_seq_len = (num_special_image_tokens.max() * (num_image_patches * num_frames - 1)) + sequence_length
```
Is there only ever 1 image token is there's an image i.e. for image inputs `max_seq_len == sequence_length`

zucchini-nlp replied:
If the prompt is only "< image >" it also works without errors. Also I added tests for inputs with interleaved and image/video only prompts.

amyeroberts commented on tests/models/video_llava/test_modeling_video_llava.py:
```diff
@@ -0,0 +1,577 @@
+        output = model.generate(**inputs, do_sample=False, max_new_tokens=20)
+
+        EXPECTED_DECODED_TEXT = [
+            'USER:  How many cats are there in the image? ASSISTANT: There are two cats in the image. hopefully, they are both sleeping.',
```
'What is this generation funny: because the addition of "hopefully" makes it seem weirdly sinister' 👀

amyeroberts commented on tests/models/video_llava/test_modeling_video_llava.py:
```diff
@@ -0,0 +1,577 @@
+        inputs = self.processor(prompt, visual_inputs=video_file, return_tensors="pt").to(torch_device, torch.float16)
+
+        output = model.generate(**inputs, max_new_tokens=900, do_sample=False)
+        EXPECTED_DECODED_TEXT = "USER:  Describe the video in details. ASSISTANT: The video features a young child sitting on a bed, holding a book and reading it. " \
```
Are we expecting the double space after USER here?

zucchini-nlp replied:
yes, the same is in Llava models when we add the special "<image>" token

amyeroberts commented on tests/models/video_llava/test_modeling_video_llava.py:
```diff
@@ -0,0 +1,577 @@
+                [
+                    32001,
+                    32001,
+                    1,
+                    15043,
+                    7084,
+                    32000,
+                    32000,
+                    32000,
+                    32000,
+                    32000,
+                    32000,
+                    32000,
+                    32000,
+                    29871,
+                    13,
+                    7900,
+                ],
+                [
+                    1,
+                    15043,
+                    7084,
+                    29901,
+                    29871,
+                    32000,
+                    32000,
+                    32000,
+                    32000,
+                    32000,
+                    32000,
+                    32000,
+                    32000,
+                    29871,
+                    13,
+                    7900,
+                ],
+            ],
+            dtype=torch.long,
+            device=torch_device,
+        )
```
lets just force the formatting off here to make it nicer to view 

```suggestion
	# fmt: off
        input_ids = torch.tensor(
            [
                [
                    32001, 32001, 1, 15043, 7084, 32000, 32000, 32000, 32000, 32000, 32000, 32000, 32000, 29871, 13, 7900
                ],
                [
                    1, 15043, 7084, 29901, 29871, 32000, 32000, 32000, 32000, 32000, 32000, 32000, 32000, 29871, 13, 7900
                ],
            ],
            dtype=torch.long,
            device=torch_device,
        )
        # fmt: on
```

amyeroberts reviewed (commented):
Thanks for adding this model. Super clean PR - was a pleasure to review! 

Mostly just small comments

nileshkokane01 commented on src/transformers/models/video_llava/modeling_video_llava.py:
```diff
@@ -0,0 +1,667 @@
+        >>> indices = np.arange(0, len(vr), len(vr) / 8).astype(int)
+        >>> frames = vr.get_batch(indices).asnumpy()
+
+        >>> inputs = processor(text=prompt, visual_inputs=clip, return_tensors="pt")
```
what is 'clip' here?

zucchini-nlp replied:
will be fixed, should be the `frames`

zucchini-nlp committed:
Update src/transformers/models/video_llava/processing_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update src/transformers/models/video_llava/image_processing_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update src/transformers/models/video_llava/image_processing_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update src/transformers/models/video_llava/image_processing_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update src/transformers/models/video_llava/image_processing_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update tests/models/video_llava/test_modeling_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp commented on src/transformers/models/video_llava/configuration_video_llava.py:
```diff
@@ -0,0 +1,134 @@
+                vocab_size=32000,
+                projection_dim=768,
+            )
+        self.vocab_size = self.vocab_size
```
Will remove. That was copied from one of the old Llava versions, but I see that `vocab_size` for VLMs is deprecated now

zucchini-nlp commented on src/transformers/models/video_llava/processing_video_llava.py:
```diff
@@ -0,0 +1,142 @@
+    def __call__(
+        self,
+        text: Union[TextInput, PreTokenizedInput, List[TextInput], List[PreTokenizedInput]] = None,
+        visual_inputs: ImageInput = None,
```
Okay, now it can be removed. It was needed before I added special token for video

zucchini-nlp commented on src/transformers/models/video_llava/modeling_video_llava.py:
```diff
@@ -0,0 +1,667 @@
+                        input_ids,
+                        attention_mask,
+                        labels,
+                        num_frames=8,
```
Yes, VideoLlava was trained and has to be used with 8 video frames. I will add it in the model docs page in "usage tips" section

amyeroberts replied:
Has this been added? Skimming I didn't spot but might have just missed

zucchini-nlp replied:
It was added in `_get_vision_features()`, after that we can never know how many frames we have

zucchini-nlp committed:
Update src/transformers/models/video_llava/image_processing_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
address review comments

zucchini-nlp committed:
failing tests

zucchini-nlp committed:
Fix vocab_size in common tests for VLMs

zucchini-nlp committed:
codestyle

zucchini-nlp committed:
Merge branch 'huggingface:main' into video_llava

amyeroberts reviewed (commented):
Thanks for the continued work on this!

zucchini-nlp committed:
Update src/transformers/models/video_llava/configuration_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update src/transformers/models/video_llava/configuration_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update src/transformers/models/video_llava/modeling_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update src/transformers/models/video_llava/modeling_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp commented on src/transformers/models/video_llava/configuration_video_llava.py:
```diff
@@ -0,0 +1,128 @@
+        if isinstance(self.vision_config, dict):
+            vision_config["model_type"] = (
+                vision_config["model_type"] if "model_type" in vision_config else "clip_vision_model"
+            )
```
I haven't seen a condition when it as missing and it's actually a copy from Llava-based models configuration file. I will raise  a warning to be explicit here for users, I guess default-clip comes from the fact that llavas are based on clip.

zucchini-nlp commented on src/transformers/models/video_llava/configuration_video_llava.py:
```diff
@@ -0,0 +1,128 @@
+        self.image_token_index = image_token_index
+        self.video_token_index = video_token_index
+        self.projector_hidden_act = projector_hidden_act
+        self.vision_feature_select_strategy = vision_feature_select_strategy
```
The `self.vision_feature_select_strategy` is checked a few lines above, in `self._get_vision_features()`. We try to get the feature and if not raise ValueError

zucchini-nlp committed:
Update docs/source/en/model_doc/video_llava.md

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update docs/source/en/model_doc/video_llava.md

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update src/transformers/models/video_llava/image_processing_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp commented on tests/test_modeling_common.py:
```diff
@@ -1762,22 +1762,32 @@ def test_resize_tokens_embeddings(self):
                 model.eval()
 
-            model_vocab_size = config.vocab_size
+            model_vocab_size = config.text_config.vocab_size if hasattr(config, "text_config") else config.vocab_size
```
okay :)

zucchini-nlp committed:
Update docs/source/en/model_doc/video_llava.md

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp commented on src/transformers/models/video_llava/processing_video_llava.py:
```diff
@@ -0,0 +1,141 @@
+              `None`).
+            - **pixel_values** -- Pixel values to be fed to a model. Returned when `images` is not `None`.
+        """
+        if images is not None or videos is not None:
+            image_kwargs = self.image_processor(images=images, videos=videos, return_tensors=return_tensors)
+        else:
+            image_kwargs = {}
+        text_inputs = self.tokenizer(
+            text, return_tensors=return_tensors, padding=padding, truncation=truncation, max_length=max_length
+        )
+
+        return BatchFeature(data={**text_inputs, **image_kwargs})
```
Hmm we cannot have no text, should be at least "< image >" to indicate that there is an image. I will remove the `text is None` locally and commit suggestion for now

zucchini-nlp committed:
Update src/transformers/models/video_llava/processing_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp commented on tests/models/video_llava/test_modeling_video_llava.py:
```diff
@@ -0,0 +1,549 @@
+if is_torch_available():
+    import torch
+else:
+    is_torch_greater_or_equal_than_2_0 = False
```
removed, was a copy from llava

zucchini-nlp committed:
Update tests/models/video_llava/test_modeling_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update tests/models/video_llava/test_modeling_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update tests/models/video_llava/test_modeling_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp commented on src/transformers/models/video_llava/image_processing_video_llava.py:
Added tests, but there is one thing to note. If we call directly the ImageProcessor class, it requires and argument `images` to be present. A workaround is to pass explicitly `images=None` for VideoLlavaImageProcessor, which I did for the tests.

I can override _call_ and to make the argument `images = None`. so that it is optional, but not sure how good is overriding _call_. Also, I do not think many ppl call image processor explicitly.

amyeroberts replied:
If the image processor takes both `images` and `videos` as input, and only one of them is required, then setting `image = None` seems reasonable

zucchini-nlp committed:
PR suggestions

zucchini-nlp committed:
fix-copies

zucchini-nlp committed:
Merge branch 'main' into video_llava

zucchini-nlp committed:
Merge branch 'huggingface:main' into video_llava

amyeroberts reviewed (approved):
Thanks for all the work adding this model - looks great!

zucchini-nlp committed:
Update src/transformers/models/video_llava/configuration_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Update src/transformers/models/video_llava/configuration_video_llava.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

zucchini-nlp committed:
Merge remote-tracking branch 'upstream/main' into video_llava

zucchini-nlp committed:
add full example in docs

zucchini-nlp commented:
Hey @LinB203, can you let us know if you can upload HF weights of VideoLlava to the organization? The model seems rready to be added to the library

LinB203 commented:
> Hey @LinB203, can you let us know if you can upload HF weights of VideoLlava to the organization? The model seems rready to be added to the library

Thanks for your great attention and employing the model in transformers. I wonder that which organization you mentioned?

zucchini-nlp commented:
@LinB203 I mean the weights that can loaded into the `transformers` model that were converted by the "convert_weights" script I added in this PR. I already have it tested and added weights to my account in the hub, but it would be nice if it was under the "LanguageBind" org. If you need to keep the current version of the state dict, you can call the new model as "LanguageBind/Video-LLaVA-7B-hf" prob

LinB203 commented:
> @LinB203 I mean the weights that can loaded into the `transformers` model that were converted by the "convert_weights" script I added in this PR. I already have it tested and added weights to my account in the hub, but it would be nice if it was under the "LanguageBind" org. If you need to keep the current version of the state dict, you can call the new model as "LanguageBind/Video-LLaVA-7B-hf" prob

I see. I will make `LanguageBind/Video-LLaVA-7B-hf` repo and upload new converted weight by your script tonight. Thank you very much.

LinB203 commented:
> @LinB203 I mean the weights that can loaded into the `transformers` model that were converted by the "convert_weights" script I added in this PR. I already have it tested and added weights to my account in the hub, but it would be nice if it was under the "LanguageBind" org. If you need to keep the current version of the state dict, you can call the new model as "LanguageBind/Video-LLaVA-7B-hf" prob

Finished.  https://huggingface.co/LanguageBind/Video-LLaVA-7B-hf/tree/main

zucchini-nlp committed:
clean-up with new model-id

zucchini-nlp committed:
[run-slow] video_llava

zucchini-nlp committed:
update docstring

zucchini-nlp committed:
Merge branch 'main' into video_llava

zucchini-nlp committed:
[run-slow] video_llava

zucchini-nlp commented:
@amyeroberts done! The last commit should trigger the slow tests after your approval. 

Note about the failing code-check, it's failing because of the `ARCHIVE_MAP` which apparently was removed for all models. So I didn't add it for VideoLlava

amyeroberts commented:
@zucchini-nlp Great! :D Could you rebase to include the upstream changes like the `ARCHIVE_MAP` removal? This should make everything green and ensure it's just that triggering the errors

zucchini-nlp committed:
Merge branch 'huggingface:main' into video_llava

zucchini-nlp committed:
remove all achive maps

zucchini-nlp commented:
The PR passed all the tests, slow tests are passing for me locally. Should be good to go

zucchini-nlp committed:
fix some tests

zucchini-nlp commented:
Just fixed one slow test, will merge when i get all green

zucchini-nlp committed:
test was supposed to be skipped for llava :)
